ext/nnstreamer/tensor_filter/tensor_filter_tensorflow_lite.cc

```diff
@@ -711,21 +721,87 @@ TFLiteInterpreter::cacheInOutTensorPtr ()
 /**
  * @brief	TFLiteCore constructor
  */
-TFLiteCore::TFLiteCore ()
+TFLiteCore::TFLiteCore (const GstTensorFilterProperties *prop)
 {
   num_threads = -1;
   accelerator = ACCL_NONE;
   delegate = TFLITE_DELEGATE_NONE;
   interpreter_sub = nullptr;
-  interpreter = new TFLiteInterpreter ();
+  shared_tensor_filter_key = NULL;
+
+  if (prop->shared_tensor_filter_key) {
+    shared_tensor_filter_key =
+        g_strdup (prop->shared_tensor_filter_key);
+    checkSharedInterpreter (prop);
+  }
+  else
+    interpreter = new TFLiteInterpreter ();
 }
 
 /**
  * @brief	TFLiteCore destructor
  */
 TFLiteCore::~TFLiteCore ()
 {
-  delete interpreter;
+  if (shared_tensor_filter_key) {
+    gpointer ref = NULL;
+    nnstreamer_filter_shared_table_lookup (shared_tensor_filter_key, &ref);
+    g_message ("[%s:%d] value: %p", __FUNCTION__, __LINE__, ref);
+    if (ref) {
+      tflite_shared_s * sTemp = (tflite_shared_s *) ref;
+
+      sTemp->referred_list = g_slist_remove (sTemp->referred_list, this);
+      if (g_slist_length (sTemp->referred_list) == 0) {
+        delete interpreter;
+        g_slist_free (sTemp->referred_list);
+        g_free (sTemp);
+        nnstreamer_filter_shared_table_remove (shared_tensor_filter_key);
+        g_free (shared_tensor_filter_key);
```

Comment:
1. I'd suggest additional API of ```nnstreamer_filter_shared_table_remove_instance ( key, this )```, assuming that any subplugin who wants "shared key" will use the same "s-list of this" or "d-list of this"
2. You need mutex associated with each key (for list removal correctness)

Reply:
3. In general, try to move most of key/value managing part to the common code so that then another subplugin starts to use this, there will be almost no additional code required.

Reply:
@kimjh12 Please follow these requirements up. We need to make it 1. thread-safe and 2. less coupling (this code of tf-lite subplugin should be able to "detach" itself from the table with a single API call.)

Reply:
With requirement 1., LINE 740 - 742 shouldn't be required (such tasks should be done by common code). The two APIs (referred_list_length and shared_table_remove) shouldn't be required. Any serialization issues should be handled inside common code and subplugins should not take care of such issues.

Reply:
I just posted a commit following the rules as far as possible.. PTAL!